### PR TITLE
feat(serve): add typescript support

### PR DESCRIPTION
**Pull request by dhruvdutt**
**What kind of change does this PR introduce?**

**Did you add tests for your changes?**

**If relevant, did you update the documentation?**

**Summary**

**Does this PR introduce a breaking change?**

**Other information**

### CODE REVIEW

File: packages/serve/index.ts [hunk 1]
```diff
@@ -94,60 +93,64 @@ function serve() {
...
 								if (packager === "yarn") {
-									spawnAction = _ => spawnYarnWithArg("--optional");
+									spawnAction = () => spawnYarnWithArg("--optional");
 								} else {
-									spawnAction = _ => spawnNPMWithArg("--save-optional");
+									spawnAction = () => spawnNPMWithArg("--save-optional");
 								}
 							}
-							return processPromise(spawnAction()).then(_ => {
+							return processPromise(spawnAction()).then(() => {
```
Comment (ev1stensberg) [inline packages/serve/index.ts]: void

File: packages/serve/index.ts [hunk 2]
```diff
@@ -94,60 +93,64 @@ function serve() {
...
 								}
 							}
-							if (depTypeAns["confirmDepType"] === "dependency") {
+							if (depTypeAns.confirmDepType === "dependency") {
 								if (packager === "yarn") {
-									spawnAction = _ => spawnYarnWithArg(" ");
+									spawnAction = () => spawnYarnWithArg(" ");
 								} else {
-									spawnAction = _ => spawnNPMWithArg("--save");
+									spawnAction = () => spawnNPMWithArg("--save");
```
Comment (ev1stensberg) [inline packages/serve/index.ts]: void

File: packages/serve/index.ts [hunk 3]
```diff
@@ -94,60 +93,64 @@ function serve() {
...
-								["devDependency", "optionalDependency", "dependency"]
-							)
+								["devDependency", "optionalDependency", "dependency"],
+							),
 						)
-						.then(depTypeAns => {
-							const packager = getRootPathModule("package-lock.json")
+						.then((depTypeAns: {
+							confirmDepType: string;
+						}) => {
```
Comment (ev1stensberg) [inline packages/serve/index.ts]: void for all empty funcs below this line

File: packages/serve/index.ts [hunk 4]
```diff
@@ -53,39 +52,39 @@ const getRootPathModule = dep => path.resolve(process.cwd(), dep);
...
 		console.log(
 			"\n",
 			chalk.red("✖ Could not find your package.json file"),
-			"\n"
+			"\n",
 		);
 		process.exit(1);
 	}
-	let packageJSON = require(packageJSONPath);
+	const packageJSON = require(packageJSONPath);
```
Comment (ev1stensberg) [inline packages/serve/index.ts]: obj

File: packages/serve/index.ts (hunk 2 above)
Comment (dhruvdutt) [reply packages/serve/index.ts]: This is a promise, hence I didn't add `_: void`. If we add it, we'll have to pass something here or use syntax `_?: void`.
https://github.com/webpack/webpack-cli/blob/617af0bddb5140c07b1a0079b377311b10dd027d/packages/serve/index.js#L150

File: packages/serve/index.ts (hunk 3 above)
Comment (dhruvdutt) [reply packages/serve/index.ts]: Please check above comment.

File: packages/serve/index.ts (hunk 2 above)
Comment (ev1stensberg) [reply packages/serve/index.ts]: `_?: void` syntax, that's normal convention across the repository.

File: packages/serve/index.ts (hunk 3 above)
Comment (ev1stensberg) [reply packages/serve/index.ts]: `_?: void` syntax

File: packages/serve/index.ts [hunk 5]
```diff
@@ -1,12 +1,11 @@
...
+import * as path from "path";
 
-const inquirer = require("inquirer");
-const path = require("path");
-const chalk = require("chalk");
-const spawn = require("cross-spawn");
-const List = require("@webpack-cli/webpack-scaffold").List;
-const processPromise = require("@webpack-cli/utils/resolve-packages")
-	.processPromise;
+import { processPromise } from "@webpack-cli/utils/resolve-packages";
```
Comment (dhruvdutt) [inline packages/serve/index.ts]: Using `utils` package here.

File: packages/serve/package.json [hunk 6]
```diff
@@ -11,8 +11,18 @@
   "license": "MIT",
   "dependencies": {
     "@webpack-cli/webpack-scaffold": "^0.0.8",
+    "@webpack-cli/utils": "^0.0.8",
```
[the remark this answers is not in the document]
Comment (dhruvdutt) [reply packages/serve/package.json]:
```
import { processPromise } from "@webpack-cli/utils/resolve-packages";
```
Check above comment. We're using utils package.

File: packages/serve/index.ts (hunk 5 above)
Comment (ev1stensberg) [reply packages/serve/index.ts]: oki

**Pull request merged**